elliott/elliottlib/cli/find_builds_cli.py

```diff
@@ -224,22 +231,89 @@ async def find_builds_cli(runtime: Runtime, advisory, default_advisory_type, bui
         else:  # attach
             erratum.attach_builds(unshipped_builds, kind)
             cdn_repos = et_data.get('cdn_repos')
-            if cdn_repos and not no_cdn_repos and kind == "image":
-                erratum.set_cdn_repos(cdn_repos)
+            if kind == 'image':
+                set_rhcos_file_meta(advisory_id)
+                if cdn_repos and not no_cdn_repos:
+                    erratum.set_cdn_repos(cdn_repos)
+
     except ErrataException as e:
-        red_print(f'Cannot change advisory {advisory}: {e}')
+        red_print(f'Cannot change advisory {advisory_id}: {e}')
         exit(1)
 
 
-def _fetch_nvrps_by_nvr_or_id(ids_or_nvrs, tag_pv_map, ignore_product_version=False, brew_session: koji.ClientSession = None):
-    session = koji.ClientSession(constants.BREW_HUB)
-    builds = brew.get_build_objects(ids_or_nvrs, session)
+def get_rhcos_nvrs_from_assembly(runtime: Runtime, brew_session: koji.ClientSession = None):
+    rhcos_config = assembly_rhcos_config(runtime.get_releases_config(), runtime.assembly)
+    build_ids_by_arch = dict()
+    nvrs = []
+
+    # Keys under rhcos_config are not necessary payload tags. One exception is `dependencies`
+    # make sure we only process payload tags
+    rhcos_payload_tags = [c['name'] for c in get_container_configs(runtime)]
+    for key, config in rhcos_config.items():
+        if key not in rhcos_payload_tags:
+            continue
+
+        for arch, pullspec in config['images'].items():
+            build_id = get_build_id_from_rhcos_pullspec(pullspec, runtime.logger)
+            if arch not in build_ids_by_arch:
+                build_ids_by_arch[arch] = set()
+            build_ids_by_arch[arch].add(build_id)
+
+    for arch, builds in build_ids_by_arch.items():
+        for build_id in builds:
+            nvr = f'rhcos-{arch}-{build_id}'
+            if brew_session.getBuild(nvr):
+                runtime.logger.info(f'Found rhcos nvr: {nvr}')
+                nvrs.append(nvr)
+            else:
+                runtime.logger.warning(f'rhcos nvr not found: {nvr}')
+    return nvrs
+
+
+def set_rhcos_file_meta(advisory_id):
+    # this assumes that the advisory is in NEW_FILES state
+
+    file_meta = errata.get_file_meta(advisory_id)
+    rhcos_file_meta = []
+    for f in file_meta:
+        # rhcos artifact file path is something like
+        # `/mnt/redhat/brewroot/packages/rhcos-x86_64/413.92.202307260246/0/images
+        # /coreos-assembler-git.tar.gz`
+
+        # title will be None if it isn't set
+        # skip if it is set
+        # skip if it is not an rhcos file
+        if f['title'] or 'rhcos' not in f['file']['path']:
+            continue
+
+        arch = None
+        for a in brew_arches:
+            if a in f['file']['path']:
+                arch = a
+                break
```

Comment:
same outcome, fewer lines:
```suggestion
        arch = next(a for a in brew_arches if a in f['file']['path'])
```
might want something that explodes if none of them matches?

Reply:
Updated